echfs-fuse.c

```diff
@@ -600,47 +588,19 @@ static int echfs_opendir(const char *dir_path,
     return 0;
 }
 
-static int echfs_fgetattr(const char *path, struct stat *stat,
+static int echfs_getattr(const char *path, struct stat *stat,
         struct fuse_file_info *file_info) {
     echfs_debug("fgetattr() on %s\n", path);
-    if (file_info->fh >= MAX_HANDLES) return -EBADF;
-    if (!handles[file_info->fh].occupied) return -EBADF;
-    struct echfs_handle_t *handle = &handles[file_info->fh];
-    struct path_result_t *path_result = handle->path_res;
-
-    stat->st_ino = path_result->target_entry + 1;
-    stat->st_nlink = 1;
-    stat->st_uid = path_result->target.owner;
-    stat->st_gid = path_result->target.group;
-    stat->st_rdev = 0;
-    stat->st_size = path_result->target.size;
-    stat->st_blksize = 512;
-    stat->st_blocks = (stat->st_size + 512 - 1) / 512;
-    stat->st_atim.tv_sec = path_result->target.atime;
-    stat->st_atim.tv_nsec = 0;
-    stat->st_mtim.tv_sec = path_result->target.mtime;
-    stat->st_mtim.tv_nsec = 0;
-    stat->st_ctim.tv_sec = path_result->target.ctime;
-    stat->st_ctim.tv_nsec = 0;
-
-    stat->st_mode = 0;
-    switch (path_result->target.type) {
-        case DIRECTORY_TYPE:
-            stat->st_mode |= S_IFDIR;
-            break;
-        case FILE_TYPE:
-            stat->st_mode |= S_IFREG;
-            break;
+    struct path_result_t *path_result;
+    if (file_info) {
```

Comment:
Why was all of this removed? Right now it seems this doesn't do anything.

Reply:
For libfuse2, there were separate functions for the same functionality: one that could work on the file handle, and one that works on the node. In libfuse3 the same function gets called in either case, but this isn't very clear from the diff